packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -122,49 +171,81 @@ export const prepareStatusManager = zone => {
       },
 
       /**
-       * Find an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`
+       * @param {CctpTxEvidence} evidence
+       */
+      isSeen(evidence) {
+        const seenKey = seenTxKeyOf(evidence);
+        return seenTxs.has(seenKey);
+      },
+
+      /**
+       * Remove and return an `ADVANCED` or `OBSERVED` tx waiting to be `SETTLED`.
        *
        * @param {NobleAddress} address
        * @param {bigint} amount
-       * @returns {boolean}
+       * @returns {Pick<PendingTx, 'status' | 'txHash'> | undefined} undefined if nothing is settleable
        */
-      hasPendingSettlement(address, amount) {
+      dequeueStatus(address, amount) {
         const key = makePendingTxKey(address, amount);
+        if (!pendingTxs.has(key)) return undefined;
         const pending = pendingTxs.get(key);
-        return !!pending.length;
+        if (!pending.length) return undefined;
+
+        const dequeueIdx = pending.findIndex(
+          x => x.status !== PendingTxStatus.Advancing,
+        );
+        if (dequeueIdx < 0) return undefined;
+
+        if (pending.length > 1) {
+          const pendingCopy = [...pending];
+          pendingCopy.splice(dequeueIdx, 1);
+          pendingTxs.set(key, harden(pendingCopy));
+        } else {
+          pendingTxs.delete(key);
+        }
+
+        const { status, txHash } = pending[dequeueIdx];
+        // TODO: store txHash -> evidence for txs pending settlement?
+        return harden({ status, txHash });
       },
 
       /**
-       * Mark an `ADVANCED` or `OBSERVED` transaction as `SETTLED` and remove it
+       * Mark a transaction as `DISBURSED`
        *
+       * @param {EvmHash} txHash
        * @param {NobleAddress} address
        * @param {bigint} amount
        */
-      settle(address, amount) {
-        const key = makePendingTxKey(address, amount);
-        const pending = pendingTxs.get(key);
-
-        if (!pending.length) {
-          throw makeError(`No unsettled entry for ${q(key)}`);
-        }
+      disbursed(txHash, address, amount) {
+        // TODO: store txHash -> evidence for txs pending settlement?
+        console.log('TODO: vstorage update', { txHash, address, amount });
```

Comment:
can wait for another PR but should at least have a TODO in this one…

define the states and valid transitions in constants, and when transitioning state verify that it's a valid transition.  like in https://github.com/Agoric/agoric-sdk/blob/2dc941f3fcc0369674fad499f0e6041e4ebbfdd9/packages/inter-protocol/src/vaultFactory/vault.js#L67-L77

Reply:
I'm adding a TODO in constants.js, but in this `disbursed(...)` case, the previous state has been dequeued, so this exo no longer has it. In fact, I'm struggling to find anywhere that such a check would be relevant.

Reply:
gotcha, maybe it doesn't make sense here with this virtual entity